src/rmp/src/delay_optimization_strategy.cpp

```diff
@@ -0,0 +1,134 @@
+extern Abc_Ntk_t* Abc_NtkMap(Abc_Ntk_t* pNtk,
+                             Mio_Library_t* userLib,
+                             double DelayTarget,
+                             double AreaMulti,
+                             double DelayMulti,
+                             float LogFan,
+                             float Slew,
+                             float Gain,
+                             int nGatesMin,
+                             int fRecovery,
+                             int fSwitching,
+                             int fSkipFanout,
+                             int fUseProfile,
+                             int fUseBuffs,
+                             int fVerbose);
+extern void Abc_FrameSetLibGen(void* pLib);
+}  // namespace abc
+
+namespace rmp {
+
+utl::UniquePtrWithDeleter<abc::Abc_Ntk_t> WrapUnique(abc::Abc_Ntk_t* ntk)
+{
+  return utl::UniquePtrWithDeleter<abc::Abc_Ntk_t>(ntk, &abc::Abc_NtkDelete);
+}
+
+void AbcPrintStats(const abc::Abc_Ntk_t* ntk)
+{
+  abc::Abc_NtkPrintStats(const_cast<abc::Abc_Ntk_t*>(ntk),
+                         false,
+                         false,
+                         false,
+                         false,
+                         false,
+                         false,
+                         false,
+                         false,
+                         false,
+                         false);
+}
+
+utl::UniquePtrWithDeleter<abc::Abc_Ntk_t> BufferNetwork(
+    abc::Abc_Ntk_t* ntk,
+    AbcLibrary& abc_sc_library)
+{
+  abc::SC_BusPars buffer_parameters;
+  memset(&buffer_parameters, 0, sizeof(abc::SC_BusPars));
+  buffer_parameters.GainRatio = 300;
+  buffer_parameters.Slew
+      = abc::Abc_SclComputeAverageSlew(abc_sc_library.abc_library());
+  buffer_parameters.nDegree = 5;
+  buffer_parameters.fSizeOnly = 0;
+  buffer_parameters.fAddBufs = false;
+  buffer_parameters.fBufPis = true;
+  buffer_parameters.fUseWireLoads = 0;
+  buffer_parameters.fVerbose = true;
+  buffer_parameters.fVeryVerbose = false;
+
+  utl::UniquePtrWithDeleter<abc::Abc_Ntk_t> buffered_network(
+      abc::Abc_SclBufferingPerform(
+          ntk, abc_sc_library.abc_library(), &buffer_parameters),
+      &abc::Abc_NtkDelete);
+  return buffered_network;
+}
+
+utl::UniquePtrWithDeleter<abc::Abc_Ntk_t> DelayOptimizationStrategy::Optimize(
+    const abc::Abc_Ntk_t* ntk,
+    utl::Logger* logger)
+{
+  auto library = static_cast<abc::Mio_Library_t*>(ntk->pManFunc);
+  // Install library for NtkMap
+  abc::Abc_FrameSetLibGen(library);
+
+  // Set up libraries for buffering
+  AbcLibraryFactory library_factory(logger);
+  library_factory.AddDbSta(sta_);
+  AbcLibrary abc_sc_library = library_factory.Build();
+  abc::Abc_SclHashCells(abc_sc_library.abc_library());
+  abc::Abc_SclLinkCells(abc_sc_library.abc_library());
+
+  AbcPrintStats(ntk);
+
+  utl::UniquePtrWithDeleter<abc::Abc_Ntk_t> current_network(
+      abc::Abc_NtkToLogic(const_cast<abc::Abc_Ntk_t*>(ntk)),
+      &abc::Abc_NtkDelete);
```

Comment:
Does Abc_NtkToLogic return a new Abc_Ntk_t?  If so it shouldn't require a non-const input.

Reply:
Abc_NtkToLogic and most ABC passes are const-able, but ABC never declares them as const. I set const on the Optimize input to at least let OpenROAD devs that the input network isn't modified. I can try to send a PR to make it const as well, but once again I don't think this should block the PR.